### PR TITLE
add memocontract plugin

### PR DESCRIPTION
**Pull request by shujip**
Thank you for your wonderful project.
I want to use depay route to call my contract, which is mint function of NFT
Please review it.

### CODE REVIEW

File: contracts/MemoworldNFTMint01.sol [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+  constructor() {}
+
+  function execute(
+    address[] calldata path,
+    uint256[] calldata amounts,
+    address[] calldata addresses,
+    string[] calldata data
+  ) external payable returns (bool) {
+    IMemoworldNFT memoContract = IMemoworldNFT(addresses[addresses.length - 1]);
```
Comment (10xSebastian) [inline contracts/MemoworldNFTMint01.sol:25]: The address here needs to be hardcoded and `MemoworldNFT` needs to be deployed prior to this plugin. It also needs to be part of this review.

File: contracts/MemoworldNFTMint01.sol [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    memoContract.mint{value: amounts[1]}(addresses[0], amounts[5], amounts[6], data[0]);
```
Comment (10xSebastian) [inline contracts/MemoworldNFTMint01.sol:27]: what information are you transporting through data[0]?

File: contracts/MemoworldNFTMint01.sol (hunk 1 above)
Comment (shujip) [reply contracts/MemoworldNFTMint01.sol:25]: Thank you for your review. OK, I will modify address to hardcoded later.
MemoworldNFT contract is on testnet (mumbai) now.  Here it is: https://mumbai.polygonscan.com/address/0xD71bD4F16391be5e108E08f2C54eac458bB9c8f7

I will deploy it to mainnet and update the address before deploy the plugin.

File: contracts/MemoworldNFTMint01.sol (hunk 2 above)
Comment (shujip) [reply contracts/MemoworldNFTMint01.sol:27]: The data[0] is just pass to a parent function "_mint". It seem just a remark for mint. I can set it a static empty string instead.

**Review by shujip — COMMENTED** [review]
Hi, I have hardcoded contract address, and update params as you mentioned

My contract is: https://polygonscan.com/address/0x4908b9381606C34EaF4ea24C01Eb66CBAB2bc749

please review for it. Thank you.

**Comment by 10xSebastian** [comment]
can you please verify and publish the contract on polygonscan.com

https://polygonscan.com/address/0x4908b9381606C34EaF4ea24C01Eb66CBAB2bc749#code

Go through the "Verify and Publish" process.

**Comment by shujip** [comment]
> can you please verify and publish the contract on polygonscan.com
> 
> https://polygonscan.com/address/0x4908b9381606C34EaF4ea24C01Eb66CBAB2bc749#code
> 
> Go through the "Verify and Publish" process.

Hi, Sebastian

    I have done Verify and Publish. Please review. Thank you.

**Pull request merged**